generators/python-v2/ast/src/PythonFile.ts

```diff
@@ -44,47 +48,99 @@ export class PythonFile extends AstNode {
     }
 
     public write(writer: Writer): void {
+        const uniqueReferences = this.deduplicateReferences();
```

Comment:
I hope to get rid of the need for this method altogether later. Right now, the data structure we use for ref tracking allows for duplicative refs, but there's no reason it needs to. We can deduplicate upon adding a new ref.

In any case, this is just a refactor to take existing logic that was there and move it into a helper so its return value can be used in two places directly below.